### ISSUE TITLE
Add ability to name breakpoints

### ISSUE THREAD

**Issue by pmorie** [derekparker/delve#109]
I would like the ability to give a breakpoint a name, especially when setting breakpoints from a file.  The use-case is that I would like to be able to be able to pass a user a breakpoint file that is set of key-value pairs (or just pairs) of breakpoint location / name that they can use locally to debug common use-cases within an app.

As an example, say a user is having some inscrutable issue that you need to have them poke around in a debugger to solve.  You can pass them a breakpoint file that tells a story and gives context for what breakpoints are what by giving a breakpoint a name, viz:

```
gh.com/someproj/pkg/registry/entity/entity_registry.go "Backing registry for entity create REST call'
```

When you hit a breakpoint with a name, should be presented to the user.

**Comment by derekparker** [comment]
That's a really neat use case. The syntax in a debug session could be something like:

```
(dlv) break <location> "[message]"
```

**Comment by pmorie** [comment]
@derekparker +1 to that syntax

**Comment by pnasrat** [comment]
Is this distinct enough to supporting arbitrary command handler for a breakpoint that you can do - eg in gdb style to merit a separate syntax here versus a general one?

```
break <location>
command <breakpoint>
print "message"
continue
end
```

**Comment by pnasrat** [comment]
Also we could have a trace command with a default print action for the message and continue - cf

https://sourceware.org/gdb/onlinedocs/gdb/Tracepoints.html
https://msdn.microsoft.com/en-us/library/ktf38f66%28v=vs.90%29.aspx

**Comment by pmorie** [comment]
@pnasrat Very interesting re: tracepoints, let's fork an issue for it.

**Comment by pmorie** [comment]
Forked #110

**Comment by aarzilli** [comment]
Implemented by 3be65a4c1f80ea71acde4b2b2195b168b14a348d
